src/BUILD.gn

```diff
@@ -81,8 +90,9 @@ target("shared_library", "v8jsi") {
     "//:v8_monolith"
   ]
 
+  defines = [ "BUILDING_V8_SHARED" ]
```

Comment:
>BUILDING_V8_SHARED [](http://example.com/codeflow?start=15&length=18)

This flag exposes all V8 API as public functions. As a result the public API surface for the DLL is huge.
It would be better if we define a separate definition for JSI API only.